src/lib/pipe_lib/strict_pipe.mli

```diff
@@ -84,6 +84,10 @@ module Writer : sig
   val to_linear_pipe : ('t, 'behavior, 'return) t -> 't Linear_pipe.Writer.t
 
   val write : ('t, _, 'return) t -> 't -> 'return
+
+  val close : (_, _, _) t -> unit
```

Comment:
you need some sort of `unclose` too right? I think you need a bool in the record that you made before